internal/graph/check.go

```diff
@@ -269,7 +269,9 @@ func union(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHandle
 		select {
 		case result := <-resultChan:
 			if result.err != nil {
-				err = result.err
+				if result.err != ErrCycleDetected {
```

Comment:
```suggestion
				if error.Is(result.err, ErrCycleDetected) {
```

Reply:
Just for consistency and to ensure any potential error wrapping upstream (not that it exists today).